chart/openfaas/templates/crd.yaml

```diff
@@ -8,9 +8,45 @@ metadata:
 spec:
   group: openfaas.com
   version: v1alpha2
+  versions:
+    - name: v1alpha2
+      served: true
+      storage: true
   names:
-    kind: Function
     plural: functions
+    singular: function
+    kind: Function
+    shortNames:
+    - fn
   scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - name
+            - image
+          properties:
+            name:
+              type: string
+              pattern: "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
```

Comment:
I think we allow upper-case - check out the regex in faas-netes for an example.

Reply:
it's the same as in faas-netes

Reply:
OK you copied it from there? Does this allow mixed-case though? I'm sure we have mixed-case deployments already supported.

Reply:
Kubernetes does not support mix-case, here is faas-netes https://github.com/openfaas/faas-netes/blob/0810d7e17bd13ceca1bfb07005fbda9de57b9078/handlers/deploy.go#L36